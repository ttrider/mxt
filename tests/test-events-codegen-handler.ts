import { codegenSetup } from "./utils";
import path from "path";
import fs from "fs";

describe("events", () => {

  const logs: string[] = [];

  afterAll(() => {

    if (logs.length > 0) {

      const filename = path.resolve(__dirname, "__snapshots__", path.basename(__filename) + ".log");
      fs.writeFileSync(filename, logs.join("\n=======================\n\n"));

    }
  });

  test("handler", () => {

    const input =
      `<template id="t01" type="mxt">
    <div id="old" style="color: \${color}" mxt.click="doClick">Hello MXT!</div>
  </template>`;

    const results = codegenSetup(input);

    //logs.push(results); 

    expect(results).toMatchSnapshot();
  });

  test("handler and propagation functions", () => {

    const input =
      `<template id="t01" type="mxt">
    <div id="old" style="color: \${color}" mxt.click="doClick" mxt.click.preventDefault mxt.click.stopPropagation mxt.click.stopImmediatePropagation>Hello MXT!</div>
  </template>`;

    const results = codegenSetup(input);

    //logs.push(results); 

    expect(results).toMatchSnapshot();
  });

  test("handler with options", () => {

    const input =
      `<template id="t01" type="mxt">
    <div id="old" style="color: \${color}" mxt.click="doClick" mxt.click.once mxt.click.passive mxt.click.capture>Hello MXT!</div>
  </template>`;

    const results = codegenSetup(input);

    logs.push(results); 

    expect(results).toMatchSnapshot();
  }); 
  
  
  test("no handler", () => {

    const input =
      `<template id="t01" type="mxt">
    <div id="old" style="color: \${color}" mxt.click.preventDefault>Hello MXT!</div>
  </template>`;

    const results = codegenSetup(input);

    //logs.push(results); 

    expect(results).toMatchSnapshot();
  }); 
});
