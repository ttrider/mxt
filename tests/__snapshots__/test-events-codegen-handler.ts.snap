// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`events handler 1`] = `
"import { autorun } from \\"mobx\\";
const t01$$template = document.createElement(\\"template\\");
t01$$template.innerHTML = \`
    <div id=\\"tagid_1\\">Hello MXT!</div>
  \`;
export function t01(data: any, host?: null | undefined | Element) {
    const component = document.importNode(t01$$template, true);
    const tagid_1$$element = component.content.getElementById(\\"tagid_1\\");
    if (!tagid_1$$element)
        throw new Error(\\"missing element: @tagid_1\\");
    tagid_1$$element.id = \\"old\\";
    tagid_1$$element.addEventListener(\\"click\\", tagid_1$$click)
    const tagid_1$$autorun = autorun(() => {
        const { color } = data;
        tagid_1$$element.setAttribute(\\"style\\", \`color: \${color}\`)
    });
    if (host)
        host.appendChild(component.content);
    return component.content;
    function tagid_1$$click(ev: Event) {
        const { doClick } = data;
        doClick(ev)
    }
}
"
`;

exports[`events handler and propagation functions 1`] = `
"import { autorun } from \\"mobx\\";
const t01$$template = document.createElement(\\"template\\");
t01$$template.innerHTML = \`
    <div id=\\"tagid_2\\">Hello MXT!</div>
  \`;
export function t01(data: any, host?: null | undefined | Element) {
    const component = document.importNode(t01$$template, true);
    const tagid_2$$element = component.content.getElementById(\\"tagid_2\\");
    if (!tagid_2$$element)
        throw new Error(\\"missing element: @tagid_2\\");
    tagid_2$$element.id = \\"old\\";
    tagid_2$$element.addEventListener(\\"click\\", tagid_2$$click)
    const tagid_2$$autorun = autorun(() => {
        const { color } = data;
        tagid_2$$element.setAttribute(\\"style\\", \`color: \${color}\`)
    });
    if (host)
        host.appendChild(component.content);
    return component.content;
    function tagid_2$$click(ev: Event) {
        const { doClick } = data;
        doClick(ev)
        ev.preventDefault()
        ev.stopPropagation()
        ev.stopImmediatePropagation()
    }
}
"
`;

exports[`events handler with options 1`] = `
"import { autorun } from \\"mobx\\";
const t01$$template = document.createElement(\\"template\\");
t01$$template.innerHTML = \`
    <div id=\\"tagid_3\\">Hello MXT!</div>
  \`;
export function t01(data: any, host?: null | undefined | Element) {
    const component = document.importNode(t01$$template, true);
    const tagid_3$$element = component.content.getElementById(\\"tagid_3\\");
    if (!tagid_3$$element)
        throw new Error(\\"missing element: @tagid_3\\");
    tagid_3$$element.id = \\"old\\";
    tagid_3$$element.addEventListener(\\"click\\", tagid_3$$click, {once, passive, capture})
    const tagid_3$$autorun = autorun(() => {
        const { color } = data;
        tagid_3$$element.setAttribute(\\"style\\", \`color: \${color}\`)
    });
    if (host)
        host.appendChild(component.content);
    return component.content;
    function tagid_3$$click(ev: Event) {
        const { doClick } = data;
        doClick(ev)
    }
}
"
`;

exports[`events no handler 1`] = `
"import { autorun } from \\"mobx\\";
const t01$$template = document.createElement(\\"template\\");
t01$$template.innerHTML = \`
    <div id=\\"tagid_4\\">Hello MXT!</div>
  \`;
export function t01(data: any, host?: null | undefined | Element) {
    const component = document.importNode(t01$$template, true);
    const tagid_4$$element = component.content.getElementById(\\"tagid_4\\");
    if (!tagid_4$$element)
        throw new Error(\\"missing element: @tagid_4\\");
    tagid_4$$element.id = \\"old\\";
    tagid_4$$element.addEventListener(\\"click\\", tagid_4$$click)
    const tagid_4$$autorun = autorun(() => {
        const { color } = data;
        tagid_4$$element.setAttribute(\\"style\\", \`color: \${color}\`)
    });
    if (host)
        host.appendChild(component.content);
    return component.content;
    function tagid_4$$click(ev: Event) {
        ev.preventDefault()
    }
}
"
`;
